eng/pipelines/templates/stages/vmr-final-join.yml

```diff
@@ -0,0 +1,79 @@
+parameters:
+# Branch of the VMR to use (to push to for internal builds)
+- name: vmrBranch
+  type: string
+  default: $(Build.SourceBranch)
+
+- name: pool_Windows
+  type: object
+  default:
+    name: $(defaultPoolName)
+    image: $(poolImage_Windows)
+    demands: ImageOverride -equals $(poolImage_Windows)
+    os: windows
+
+- name: mainVertical
+  type: string
+  default: Windows_x64
+
+- name: azureDevOpsOrg
+  type: string
+  default: dnceng
+
+- name: azureDevOpsProject
+  type: string
+  default: internal
+    
+stages:
+- stage: VMR_Final_Join
+  displayName: VMR Final Join
+  dependsOn: VMR_Vertical_Build
+  variables:
+  - ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+    - group: Publish-Build-Assets
+  - ${{ else }}:
+    - name: dn-bot-dnceng-build-rw-code-rw
+      value: ''
+
+  - template: ../variables/vmr-build.yml
+    parameters:
+      vmrBranch: ${{ parameters.vmrBranch }}
+
+  jobs:
+  - job: VMR_Final_Join
+    pool: ${{ parameters.pool_Windows }}
+
+    templateContext:
+      outputs:
+      - output: pipelineArtifact
+        path: $(Build.ArtifactStagingDirectory)/artifacts
+        artifact: JoinedArtifacts
+        condition: succeededOrFailed()
+
+    steps:
+    - checkout: self
+
+    - task: DownloadPipelineArtifact@2
+      inputs:
+        artifactName: 'VerticalManifests'
+        targetPath: $(Build.ArtifactStagingDirectory)/VerticalManifests
+
+    - script: $(Build.SourcesDirectory)/build.cmd 
+        -ci
+        /p:DotNetBuildPass=final
+        /p:VerticalManifestsPath=$(Build.ArtifactStagingDirectory)/VerticalManifests
+        /p:MainVertical=$(parameters.mainVertical)
+        /p:BuildId=$(Build.BuildId)
+        /p:AzureDevOpsToken=$(dn-bot-dnceng-build-rw-code-rw)
+        /p:AzureDevOpsOrg=$(parameters.azureDevOpsOrg)
+        /p:AzureDevOpsProject=$(parameters.azureDevOpsProject)
+        /p:TmpFolder=$(Agent.TempDirectory)
+        /p:OutputFolder=$(Build.ArtifactStagingDirectory)/artifacts
+      displayName: 'VMR Final Join'
+
+    - ${{ if or(ne(variables['System.TeamProject'], 'internal'), eq(variables['Build.Reason'], 'PullRequest')) }}:
+      - publish: $(Build.ArtifactStagingDirectory)/artifacts
+        artifact: JoinedArtifacts
+        displayName: Publish Artifacts
+        condition: succeededOrFailed()
+        continueOnError: true
```

Comment:
I wonder if we could just use the job/vmr-build.yml template instead of having all this. I think that job template already supports passing in additional msbuild parameters.

Reply:
yeah it does feel like eventually we'll be doing the same thing in the `Joint builds`, download Manifests, run build.cmd to download artifacts and build, and finally publish some things. I think tho it might be easier to do that at a later point

Reply:
I gave this some thought and I don't think it's the correct thing to do, as we'll be kind of mixing things. With the newly renamed stage `VMR Post Build` it doesn't feel right to call `vmr-build`. Instead, I created a steps template that we can call from the `vmr-build` job, when the `dotNetBuildPass` isn't 1 to prepare artifacts for the joint build